src/lib.rs

```diff
@@ -101,9 +146,21 @@ impl PyEvtxParser {
     }
 }
 
+
 #[pyproto]
 impl PyIterProtocol for PyEvtxParser {
-    fn __iter__(slf: PyRefMut<Self>) -> PyResult<Py<PyEvtxParser>> {
+    fn __iter__(mut slf: PyRefMut<Self>) -> PyResult<PyRecordsIterator> {
+        slf.records()
+    }
+    fn __next__(mut slf: PyRefMut<Self>) -> PyResult<Option<PyObject>> {
```

Comment:
Raise an exception instead of `unimplemented()`, this would segfault the python interpreter.